packages/hooks/src/Provider.tsx

```diff
@@ -48,15 +68,49 @@ export const Provider: React.FC<ProviderProps> = ({ children, network, infuraId
   const [connection, setConnection] = React.useState<any>();
 
   const connectWallet = React.useCallback(async () => {
-    const web3Modal = new Web3Modal({
-      providerOptions: {
-        walletconnect: {
-          package: WalletConnectProvider,
-          options: {
-            infuraId,
+    // Coinbase walletLink init
+    const walletLink = new WalletLink({
+      appName: 'coinbase',
+    });
+
+    // WalletLink provider
+    const walletLinkProvider = walletLink.makeWeb3Provider(
+      `https://eth-mainnet.alchemyapi.io/v2/${infuraId}`,
+      1
+    );
+
+    const defaulProviderOptions = {
+      walletconnect: {
+        package: WalletConnectProvider,
+        options: {
+          bridge: 'https://polygon.bridge.walletconnect.org',
+          infuraId,
+          rpc: {
+            1: `https://eth-mainnet.alchemyapi.io/v2/${infuraId}`, // mainnet // For more WalletConnect providers: https://docs.walletconnect.org/quick-start/dapps/web3-provider#required
+            42: infuraId,
```

Comment:
Wouldn't we need to append a link to the `infuraId` here? Just like we did for mainnet?

Reply:
Thanks for pointing it out. It just went out of my mind. Adding that.